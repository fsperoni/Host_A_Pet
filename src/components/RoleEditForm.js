import React, { useState } from "react";
import Alert from "./Alert";
import HostAPetApi from "./Api";

/** Edit Role form. */

const RoleEditForm = ({role, roles, setRoles, setShowEditForm}) => {
  const [formData, setFormData] = useState({
    name: role.name
  });
  const [formErrors, setFormErrors] = useState([]);
  

  /** Handle form submit */
  const handleSubmit = async (evt) => {
    evt.preventDefault();
    const id = parseInt(evt.target.children[0].value);
    try {
      const result = await HostAPetApi.updateRole(id, formData)
      const newRoles = roles.filter(role => role.id !== id);
      setRoles([...newRoles, result.role])
      setFormErrors([]);
      setShowEditForm(false);
    } catch (err) {
      setFormErrors(err);
    }
  }

  /** Handle form field changes */
  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData(data => ({ ...data, [name]: value }));
    setFormErrors([]);
  }

  /** Handle form cancel button */
  const handleCancel = (evt) => {
    evt.preventDefault();
    setFormErrors([]);
    setShowEditForm(false);
  }

  return (
    <div className="RoleEditForm mt-3">
      <div className="container">
        <h2 className="mb-3">Update role</h2>
        <div className="card">
          <div className="card-body">
            <form onSubmit={handleSubmit}>
              <div className="form-group">
                <label>Name:</label>
                <input name="name" className="form-control" onChange={handleChange}
                  value={formData.name}
                />
              </div>
              {formErrors.length>0 && <Alert type="danger" messages={formErrors} />}
              <div className="container mt-4">
              <button 
                className="btn btn-primary btn-block"
                onClick={handleSubmit}
              > <data value={role.id} />
                Save
              </button>
              <button
                className="btn btn-warning btn-block ms-4"
                onClick={handleCancel}
              >
                Cancel
              </button>
            </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  );
}

export default RoleEditForm;