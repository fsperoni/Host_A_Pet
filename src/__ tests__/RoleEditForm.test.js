import { render } from "@testing-library/react";
import RoleEditForm from "../components/RoleEditForm";

const {
  roles, roleForm, 
} = require("./_testCommon");

// smoke test
it('renders without crashing', function () {
  render(
        <RoleEditForm role={roleForm()} setRoles={function () { }}
        setShoweditForm={function () {}} roles={roles()} />
  );
});

//snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(
        <RoleEditForm role={roleForm()} setRoles={function () { }}
        setShoweditForm={function () {}} roles={roles()} />

  );
  expect(asFragment()).toMatchSnapshot();
});

it("should have 'Update role' in the component.", function () {
  const { getByText } = render(
        <RoleEditForm role={roleForm()} setRoles={function () { }}
        setShoweditForm={function () {}} roles={roles()} />
  );
  expect(getByText("Update role")).toBeInTheDocument();
});
