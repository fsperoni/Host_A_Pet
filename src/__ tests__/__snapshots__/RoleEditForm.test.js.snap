// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="RoleEditForm mt-3"
  >
    <div
      class="container"
    >
      <h2
        class="mb-3"
      >
        Update role
      </h2>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <form>
            <div
              class="form-group"
            >
              <label>
                Name:
              </label>
              <input
                class="form-control"
                name="name"
                required=""
                value="Pet Owner"
              />
            </div>
            <div
              class="container mt-4"
            >
              <button
                class="btn btn-primary btn-block"
              >
                 
                <data
                  value="2"
                />
                Save
              </button>
              <button
                class="btn btn-warning btn-block ms-4"
              >
                Cancel
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
